core/crypto/hash/low_level.odin

```diff
@@ -0,0 +1,287 @@
+package crypto_hash
+
+import "core:crypto/sha2"
+import "core:crypto/sha3"
+import "core:crypto/shake"
+import "core:crypto/sm3"
+
+import "core:crypto/legacy/md5"
+import "core:crypto/legacy/sha1"
+
+// Algorithm is the algorithm identifier associated with a given Context.
+Algorithm :: enum {
+	Invalid,
+	SHA224,
+	SHA256,
+	SHA384,
+	SHA512,
+	SHA512_256,
+	SHA3_224,
+	SHA3_256,
+	SHA3_384,
+	SHA3_512,
+	SHAKE_128,
+	SHAKE_256,
+	SM3,
+	Insecure_MD5,
+	Insecure_SHA1,
+}
+
+// ALGORITHM_NAMES is the Algorithm to algorithm name string map.
+ALGORITHM_NAMES := map[Algorithm]string {
```

Comment:
This could just be an enumerated array, why does this need to be a `map`?
`ALGORITHM_NAMES := [Algorithm]string { ... }`